repository/filtering.go

```diff
@@ -0,0 +1,74 @@
+// Package repository provides filter model
+package repository
+
+import (
+	"database/sql"
+
+	"github.com/google/uuid"
+)
+
+type Postgres struct {
+	db *sql.DB
+}
+
+// NewPostgres returns Postgres with db
+func NewPostgres(db *sql.DB) *Postgres {
+	return &Postgres{db: db}
+}
+
+type FiltersGUD interface {
+	GetFilters() ([]Filter, error)
+	UpdateFilter() error
+	DeleteFilter() error
+}
+
+const (
+	addQuery    = "INSERT INTO FILTERS(name,type,configutation,reversed,user_id) VALUES ($1,$2,$3,$4,$5)"
+	getQuery    = "SELECT * FROM FILTERS WHERE user_id=$1"
+	updateQuery = "UPDATE FILTERS SET (type,configutation,reversed) = ($1,$2,$3) WHERE name = $4"
+	deleteQuery = "DELETE FROM FILTERS WHERE name=$1"
+)
+
+// AddFilter adds new filter to database
+func (p *Postgres) AddFilter(filter *Filter) error {
+	_, err := p.db.Exec(addQuery, filter.Name, filter.Type, filter.Configuration, filter.Reversed, filter.UserID)
+	return err
+}
+
+// GetFilters returns set of filters relevant to user
+func (p *Postgres) GetFilters(userID uuid.UUID) ([]Filter, error) {
+	filtersSet := []Filter{}
+	r, err := p.db.Query(getQuery, userID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for r.Next() {
+		f := Filter{}
+		if err := r.Scan(&f); err != nil {

```

Comment:
Does this Scan work?

Reply:
Oh, I was thinking about jmoiron/sqlx importing, but at least decided to put into struct and haven't  corrected that. I should pay much more attention to my code(( Thank you